scene/3d/physics_body.h

```diff
@@ -346,4 +347,73 @@ class KinematicCollision : public Reference {
 	KinematicCollision();
 };
 
+class PhysicalBone : public PhysicsBody {
+
+	GDCLASS(PhysicalBone, PhysicsBody);
+
+	Skeleton *parent_skeleton;
+	bool static_body;
+	bool _internal_static_body;
+	bool simulate_physics;
+	bool _internal_simulate_physics;
+	int bone_id;
+
+	String bone_name;
+	real_t bounce;
+	real_t mass;
+	real_t friction;
+	real_t gravity_scale;
+
+protected:
+	bool _set(const StringName &p_name, const Variant &p_value);
+	bool _get(const StringName &p_name, Variant &r_ret) const;
+	void _get_property_list(List<PropertyInfo> *p_list) const;
+	void _notification(int p_what);
+	void _direct_state_changed(Object *p_state);
+
+	static void _bind_methods();
+
+private:
+	static Skeleton *get_skeleton_parent(Node *p_parent);
+
+public:
+	void set_static_body(bool p_static);
+	bool is_static_body();
+
+	void set_simulate_physics(bool p_simulate);
+	bool get_simulate_physics();
+	bool is_simulating_physics();
+
+	void set_bone_name(const String &p_name);
+	const String &get_bone_name() const;
+
+	void set_mass(real_t p_mass);
+	real_t get_mass() const;
+
+	void set_weight(real_t p_weight);
+	real_t get_weight() const;
+
+	void set_friction(real_t p_friction);
+	real_t get_friction() const;
+
+	void set_bounce(real_t p_bounce);
+	real_t get_bounce() const;
+
+	void set_gravity_scale(real_t p_gravity_scale);
+	real_t get_gravity_scale() const;
+
+	PhysicalBone();
+	~PhysicalBone();
+
```

Comment:
While not generally using joints if you want with this, these bones should include the same information that will be used for IK. This is, an option between:
* Joint Disabled
* A Regular pin
* A Cone twist
* A Hinge

and relevant options.  The joint would be created by this bone and attachment between it and the parent bone, as well as collision exception, would be handled in this class.

Reply:
Will also need an offset from the center of mass.